pkg/cloud/azure/services/internalloadbalancers/service.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internalloadbalancers
```

Comment:
why do we want a separate package for `internalloadbalancers` ?

Reply:
Each service has a purpose, defines a target behaviour rather than mimicking azure service itself, agreed that the code could be 50% same, its better to keep it in different packages to test the behaviour.

Reply:
I'd like loadbalancers to be consolidated into a single package. The requesting code path can do something like:
- Specify `NetworkSpec`
- NetworkSpec will hold something like, `NetworkType: {public,private}` or `defaultLoadBalancerType: {public,internal|private}`

From there, load balancer methods should take that as a parameter and have:
```go
func ReconcileLoadBalancer(type) {
  ...
  case: "public":
    ...
  case: "private|internal"
    ...
}
```

Let's minimize duplication of code where we can, as the probe, backend pool, frontendIp logic will largely be the same as well.

Reply:
Branches are the cause of most bugs in software, i would rather separate the implementation in separate files and maintain duplicity if any. Looking at the difference between the 2 (as of now which can change in future)

__Common:__
  * Load Balancing rules
  * Probes
  * Get 
  * Delete

__Differences:__

__InternalLoadBalancers:__
  * Static Address
  * Uses Master Subnet

__PublicLoadBalancers:__
  * Dynamic
  * Public IP Address
  * Nat Rules

There is quite a bit of difference, but to save 25-30% code duplication is not worth sacrificing the code readability. 

May be in future it would be nice to separate the configuration into separate file, would rather postpone to future when this is absolutely required.